src/w3d/renderer/streak.cpp

```diff
@@ -13,14 +14,528 @@
  *            LICENSE
  */
 #include "streak.h"
+#include "coltest.h"
+#include "w3d.h"
+
+StreakLineClass::StreakLineClass() : m_maxSubdivisionLevels(0), m_normalizedScreenArea(0), m_unk(nullptr) {}
+
+StreakLineClass::StreakLineClass(const StreakLineClass &src) :
+    m_maxSubdivisionLevels(src.m_maxSubdivisionLevels),
+    m_unk(src.m_unk),
+    m_normalizedScreenArea(src.m_normalizedScreenArea),
+    m_pointLocations(src.m_pointLocations),
+    m_pointColors(src.m_pointColors),
+    m_pointWidths(src.m_pointWidths),
+    m_lineRenderer(src.m_lineRenderer),
+    m_streakRenderer(src.m_streakRenderer)
+{
+}
+
+StreakLineClass &StreakLineClass::operator=(const StreakLineClass &that)
+{
+    if (this != &that) {
+        RenderObjClass::operator=(that);
+        m_maxSubdivisionLevels = that.m_maxSubdivisionLevels;
+        m_normalizedScreenArea = that.m_normalizedScreenArea;
+        m_pointLocations = that.m_pointLocations;
+        m_pointColors = that.m_pointColors;
+        m_pointWidths = that.m_pointWidths;
+        m_lineRenderer = that.m_lineRenderer;
+        m_streakRenderer = that.m_streakRenderer;
+        m_unk = that.m_unk;
+    }
+
+    return *this;
+}
+
+StreakLineClass::~StreakLineClass() {}
+
+void StreakLineClass::Reset_Line()
+{
+    m_lineRenderer.Reset_Line();
+}
+
+void StreakLineClass::Set_Locs(unsigned int count, Vector3 *locs)
+{
+    captainslog_assert(count >= 2 && locs != nullptr);
+    m_pointLocations.Delete_All();
+    Invalidate_Cached_Bounding_Volumes();
+    if (count != 0 && locs != nullptr) {

```

Comment:
`count != 0` check is obsolete, because `i < count` checks it next line.

In several functions.